sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/WrapperSdkExceptionManager.java

```diff
@@ -0,0 +1,156 @@
+package com.microsoft.azure.mobile.crashes;
+
+import com.microsoft.azure.mobile.crashes.utils.ErrorLogHelper;
+import com.microsoft.azure.mobile.utils.MobileCenterLog;
+import com.microsoft.azure.mobile.utils.storage.StorageHelper;
+
+import java.io.File;
+import java.util.UUID;
+import java.util.Map;
+import java.util.HashMap;
+import java.io.IOException;
+
+public class WrapperSdkExceptionManager {
+    /**
+     * File extension for data files created by this class.
+     */
+    private static final String dataFileExtension = ".dat";
+
+    /**
+     * In-memory storage to be set by a wrapper SDK that contains the data it needs to save
+     */
+    private static byte[] mManagedExceptionData = null;
+
+    /**
+     * Contains wrapper SDK data that has been loaded into memory
+     */
+    private static Map<String, byte[]> mManagedExceptionDataContainer = new HashMap<>();
+
+    /**
+     * Check if there is managed exception data in memory
+     *
+     * @return true if there is exception data in memory
+     */
+    public static boolean hasManagedExceptionData() {
+        return mManagedExceptionData != null;
+    }
+
+    /**
+     * Store wrapper exception data in memory. This should only be used by a wrapper SDK.
+     *
+     * @param data  byte array containing the data a wrapper SDK needs to save
+     */
+    public static void setManagedExceptionData(byte[] data) {
```

Comment:
See my other comment, we should not do this. Let xamarin keep in memory and call `save`.